Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-18.ca-central-1.compute.internal
Sha: b10fed7596a908988f2b82aeef2d943643234648
Number of Error(s): 5
/home/centos/smoketest/PR-14793/HPCC-Platform/system/jlib/jsecrets.cpp:744:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14793/HPCC-Platform/system/jlib/jsecrets.cpp:748:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:492: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-04-07_22-19-18
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   333 sec (00:05:33)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   342 sec (00:05:42)
time logs end

[Build error:](https://gist.github.com/88a8261809f1cea298b9324e24ccb08b)
Build log [Build log](https://gist.github.com/HPCCSmoketest/88a8261809f1cea298b9324e24ccb08b#file-result-21-04-07-22-13-23-log)

--- a/system/jlib/jsecrets.cpp
+++ b/system/jlib/jsecrets.cpp
@@ -557,10 +557,15 @@ static const char *ensureSecretDirectory()
     return secretDirectory;
 }
 
+static StringBuffer &buildSecretPath(StringBuffer &path, const char *category, const char * name)
+{
+    return addPathSepChar(path.append(ensureSecretDirectory())).append(category).append(PATHSEPCHAR).append(name).append(PATHSEPCHAR);
+}
+
 static IPropertyTree *loadLocalSecret(const char *category, const char * name)
 {
     StringBuffer path;
-    addPathSepChar(path.append(ensureSecretDirectory())).append(category).append(PATHSEPCHAR).append(name).append(PATHSEPCHAR);
+    buildSecretPath(path, category, name);
     Owned<IDirectoryIterator> entries = createDirectoryIterator(path);
     if (!entries || !entries->first())
         return nullptr;
@@ -727,8 +732,10 @@ void initSecretUdpKey()
     if (udpKeyInitialized)
         return;
 
-#if defined(_CONTAINERIZED) && defined(_USE_OPENSSL)
-    BIO *in = BIO_new_file("/opt/HPCCSystems/secrets/certificates/udp/tls.key", "r");
+//can find alternatives for old openssl in the future if necessary
+#if defined(_USE_OPENSSL) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
+    StringBuffer path;
+    BIO *in = BIO_new_file(buildSecretPath(path, "certificates", "udp").append("tls.key"), "r");
     if (in == nullptr)
         return;
     EC_KEY *eckey = PEM_read_bio_ECPrivateKey(in, nullptr, nullptr, nullptr);
@@ -769,7 +776,7 @@ IPropertyTree *queryMtlsSecretInfo(const char *name)
     StringBuffer filepath;
     StringBuffer secretpath;
 
-    addPathSepChar(secretpath.append(ensureSecretDirectory())).append("certificates").append(PATHSEPCHAR).append(name).append(PATHSEPCHAR);
+    buildSecretPath(secretpath, "certificates", name);
 
     filepath.set(secretpath).append("tls.crt");
     if (!checkFileExists(filepath))

